File: src/02.string/05.subsequence/01.392.go
package subsequence

func IsSubsequence(s string, t string) bool {
	n, m := len(t), len(s)
	sIndex := 0

	for i := 0; i < n && sIndex < m; i++ {
		if s[sIndex] == t[i] {
			sIndex++
		}
	}

	return sIndex == m
}

File: src/02.string/05.subsequence/01.392.ts
export function isSubsequence(s: string, t: string): boolean {
    const n = t.length;
    const m = s.length;
    let sIndex = 0;
    for (let i = 0; i < n && sIndex < m; i++) {
        if (s[sIndex] === t[i]) {
            sIndex++;
        }
    }

    return sIndex === m;
}
